S/slow5lib/build_tarballs.jl

```diff
@@ -0,0 +1,43 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "slow5lib"
+version = v"1.3.1"
+
+sources = [
+    GitSource("https://github.com/hasindu2008/slow5lib.git", "050c09868696d98735152bf0abb3de766f5ab5cb"),
+    DirectorySource("./patches")
+]
+
+script = raw"""
+if [[ "${target}" == *-apple-darwin* ]]; then
+    actual_lib_dir="${WORKSPACE}/$(readlink ${WORKSPACE}/destdir)/lib"
+    export LDFLAGS="-L${actual_lib_dir} -lz"
+else
+    export LDFLAGS="-L${libdir}"
+fi
+cd $WORKSPACE/srcdir
+atomic_patch -p1 fix_cmakelists_for_slow5lib.patch
+cd slow5lib/
+cmake -B build -DCMAKE_INSTALL_PREFIX=$prefix -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} -DCMAKE_BUILD_TYPE=Release
+cmake --build build --parallel ${nproc}
+cmake --install build
+cp $WORKSPACE/srcdir/slow5lib/build/libslow5.${dlext} $WORKSPACE/destdir/lib
+cp -r $WORKSPACE/srcdir/slow5lib/include/slow5 $WORKSPACE/destdir/include
+install_license ${WORKSPACE}/srcdir/slow5lib/LICENSE
+"""
+
+platforms = filter(plt -> plt.tags["os"] != "windows" && plt.tags["arch"] != "riscv64",
+                   supported_platforms())
```

Comment:
Better to use the relevant helpers to query the platform objects instead of assuming specifics about the internal implementation. Also, separating the filter conditions in the past has been useful, and most other recipes follow a patter like that, making maintenance a tad bit easier:
```suggestion
platforms = supported_platforms()
filter!(!Sys.iswindows, platforms)
filter!(p -> arch(p) != "riscv64", platforms)
```